File: Examples/SequenceDiagram/Java/PaymentService.java
package shop;

/** Charges an order by delegating to the payment gateway. */
public final class PaymentService {
    private final PaymentGateway gateway;

    public PaymentService(PaymentGateway gateway) {
        this.gateway = gateway;
    }

    public void charge() {
        gateway.authorize();
    }
}

File: Examples/SequenceDiagram/Python/PaymentService.py
from PaymentGateway import PaymentGateway


class PaymentService:
    """Charges an order by delegating to the payment gateway."""

    def __init__(self, gateway: PaymentGateway):
        self.gateway: PaymentGateway = gateway

    def charge(self) -> None:
        self.gateway.authorize()
